BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -0,0 +1,400 @@
+        case "HKBiologicalSex":
+            return val as? NSNumber
+            
+        case "HKQuantity":
+            return val as? NSNumber
+            
+        default:
+            return nil
+        }
+    }
+    
+    func commonJsonValueSetter(value: SBBJSONValue?) {
+        guard value != nil else {
+            self.value = nil
+            return
+        }
+        
+        switch self.itemType {
+        case "String":
+            guard let val = value as? String else { return }
+            self.value = val
+            
+        case "Number":
+            guard let val = value as? NSNumber else { return }
+            self.value = val
+            
+        case "Bool":
+            guard let val = value as? Bool else { return }
+            self.value = val
+            
+        case "Date":
+            guard let stringVal = value as? String,
+                    let dateVal = NSDate(iso8601String: stringVal)
+                else { return }
+            self.value = dateVal
+            
+        case "HKBiologicalSex":
+            guard let val = value as? HKBiologicalSex else { return }
+            self.value = val
+            
+        case "HKQuantity":
+            guard let val = value as? NSNumber else { return }
+            self.value = val
+            
+        default:
+            break
+        }
+    }
+    
+    func commonDemographicJsonValue() -> SBBJSONValue? {
+        guard let jsonVal = self.commonJsonValueGetter() else { return nil }
+        if self.itemType == "HKBiologicalSex" {
+            return (self.value as? HKBiologicalSex)?.demographicDataValue
+        }
+        
+        return jsonVal
+    }
+    
+    func commonCheckTypeCompatible(newValue: Any?) -> Bool {
+        guard newValue != nil else { return true }
+        
+        switch self.itemType {
+        case "String":
+            return newValue as? String != nil
+            
+        case "Number":
+            return newValue as? NSNumber != nil
+            
+        case "Bool":
+            return newValue as? NSNumber != nil
+            
+        case "Date":
+            return newValue as? NSDate  != nil
+            
+        case "HKBiologicalSex":
+            return newValue as? HKBiologicalSex != nil
+            
+        case "HKQuantity":
+            return newValue as? NSNumber != nil
+            
+        default:
+            return false
+        }
+    }
+}
+
+class SBAProfileItemBase: NSObject, SBAProfileItem {
+    /**
+     The value property is used to get and set the profile item's value in whatever internal data
+     storage is used by the implementing class.
+     */
+    open var value: Any?
+
+    fileprivate let sourceDict: [AnyHashable: Any]
+    
+    open var key: String {
+        get {
+            return sourceDict["key"]! as! String
+        }
+    }
+    
+    open var sourceKey: String {
+        get {
+            return sourceDict["sourceKey"] as! String? ?? self.key
```

Comment:
indeed it should--Xcode's Swift suggestions when I get it wrong seem to be wrong as often as right :/